### ISSUE TITLE
bug on format class where clause

### ISSUE THREAD

**Issue by leptonyu** [commercialhaskell/hindent#482]
```Haskell
class Monad m => MonadHello m where
  hello :: m Text
```

after format
```Haskell
class Monad m =>
      MonadHello m
  | 
  where
  hello :: m Text
```

version hindent 5.2.4.1

**Comment by JustusAdam** [comment]
For me it actually doesn't insert the `|`. (Version `5.2.5`)

I however have a related problem. If there's a documentation comment before an associated type (don't know if the same is true for methods) after formatting the doc comment comes **before** the `where`, which is a parse error for haddock.

```hs
class AClass c where
    -- |  Some documentation
    type MyType c
```

formats to

```hs
class AClass c
    -- |  Some documentation
    where
    type MyType c
```

**Comment by chrisdone** [comment]
Seems fixed on 5.2.5.
